tracer/src/Datadog.Trace/Agent/TraceSamplers/RareSampler.cs

```diff
@@ -30,7 +36,16 @@ internal class RareSampler : ITraceChunkSampler
         /// <returns>true when a rare span is found, false otherwise</returns>
         public bool Sample(ArraySegment<Span> trace)
         {
-            if (SamplingHelpers.IsKeptBySamplingPriority(trace))
```

Comment:
Why are we switching the condition from "if sampling priority is 1 or 2" to "if the sampling priority was set manually"? I'm not saying it's wrong, just trying to understand.

Also, the comment above needs updating:
```
1) If the sampling priority is > 0, only update the seen spans and return false.
```